### DISCUSSION TITLE
Order of processing?

### DISCUSSION THREAD

**Discussion by wn2000** [abraunegg/onedrive#1314] [Q&A] [answered]
I'm trying to migrate a large OneDrive Business Shared Folder to local drive, so I can delete the remote files after that.

If I see the program has moved on to download files in a new subfolder, can I assume it has finished downloading all previous folders? Asking as I do not want to wait until everything downloaded then delete remote folders, that would take a long time.

Thanks.

**Comment by abraunegg** [comment, marked as answer]
@wn2000 
> If I see the program has moved on to download files in a new subfolder, can I assume it has finished downloading all previous folders?

This depends on how the OneDrive API is sending the data regarding the files stored online. Most likely this is the case, that when the client moves onto a new folder, the prior folder is complete.

It would be advisable however to *wait* until all files are downloaded, that you validate that all files are present locally, confirm that they are with the right checksum / non-corrupt, before *deleting* anything online.
